src/common.jl

```diff
@@ -20,13 +20,6 @@ const IntegerMatrix{T<:Integer} = AbstractArray{T,2}
 
 const RealFP = Union{Float32, Float64}
 
-## conversion from real to fp types
-
-fptype(::Type{T}) where {T<:Union{Float32,Bool,Int8,UInt8,Int16,UInt16}} = Float32
```

Comment:
`float(Int8) === Float64`, so this PR changes the behavior. This warrants a discussion at least. (Same for place where `typeof(zero(eltype(X))/1)` is used.

A less disruptive fix would be to just add a fallback `fptype(::Type{T}) where {T} = float(T)`.

Reply:
IMO the behaviour of `fptype` was completely unexpected and inconsistent with the default promotion logic in Julia that I've seen in other places (e.g., in `mean`). I expect floating point numbers of type `Float64` in all the cases `Bool`, `Int8` etc. since that is the default floating point number type in Julia and only of type `Float32` if the input type contains floating point numbers of type `Float32`.

Additionally, users can still provide output arrays and hence obtain results with floating point numbers of type `Float32` even in these cases.

In any case, it seems it is an additional argument for https://github.com/JuliaStats/StatsBase.jl/pull/566.

Reply:
OK, everybody agrees with the change, so we'll tag a new minor release.